app/controllers/internal/events_controller.rb

```diff
@@ -26,8 +27,7 @@ def show
     def new
       if params[:duplicate]
         @event = get_event_by_id(params[:duplicate])
-        @event.id = nil
-        @event.readable_id = nil
+        @event.assign_attributes(NILIFY_ON_DUPLICATE.to_h { |attribute| [attribute, nil] })
```

Comment:
Potentially doing something a bit weird here.

Tried all of these methods https://makandracards.com/makandra/42641-different-ways-to-set-attributes-in-activerecord

and the only one that worked was `assign_attributes`

Reply:
Just wondering, could this be moved to a model method?

Perhaps something along the lines of:

```
def self.copy(existing_event)
  new(existing_event).tap do |e|
    e.starts_at = nil
    e.ends_at = nil
    ...
  end
end
```

That'd make duplicating easy to test on its own and keep everything in one place?

Reply:
Looking at some more duplicate stuff now, so I'll bear this in mind